src/core/BufferAttribute.js

```diff
@@ -327,7 +333,12 @@ class BufferAttribute {
 
 		this.array[ index + 0 ] = x;
 		this.array[ index + 1 ] = y;
-		this.array[ index + 2 ] = z;
+
+		if ( this.itemSize > 2 ) {
+
+			this.array[ index + 2 ] = z;
+
+		}
```

Comment:
It think we should also update `setZ()` in a similar fashion for consistency reasons.

Reply:
And finally the methods of `InterleavedBufferAttribute` should get the same updates.